stripe-core/src/main/java/com/stripe/android/core/reactnative/ReactNativeSdkInternal.kt

```diff
@@ -0,0 +1,8 @@
+package com.stripe.android.core.reactnative
+
+@RequiresOptIn(
+    level = RequiresOptIn.Level.ERROR,
+    message = "This API should only be used internally by the react-native sdk."
+)
+@Retention(AnnotationRetention.BINARY)
+annotation class ReactNativeSdkInternal
```

Comment:
We should mark all the new public APIs with @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)

You can check this locally via `./gradlew apiCheck`.